tools/jit/gen_jit_dispatch.py

```diff
@@ -91,20 +99,24 @@ def gen_jit_dispatch(declarations, out):
     def is_tensor_arg(arg):
         return arg['simple_type'] in {'Tensor', 'TensorList'}
 
+    def is_sized_intlist_arg(arg):
+        return (arg['simple_type'] == 'IntList') and ('size' in arg)
+
     def get_invocation(decl, args):
         if 'namespace' in decl['method_of']:
             return CALL_NAMESPACE.substitute(name=decl['name'], args=args)
         else:
             return CALL_METHOD.substitute(name=decl['name'], first=args[0], args=args[1:])
 
-    def emit_decl_variant(decl, is_positional_arg, has_tensorlist):
+    def emit_decl_variant(decl, is_positional_arg, has_tensorlist, indices_to_replace=None):
```

Comment:
As far as I can see this only affects the attribute case, but I don't think this is necessary. Instead of going through the powerset and blowing up the number of cases we could just alter the way we parse attribute values, similarly to what we do with positional args. This has both better complexity, and doesn't require this extra argument with pretty annoying semantics.

Reply:
Can you please still change this one? That's the only blocker for this PR.

Reply:
I'm really sorry, could you clarify that a bit?

You'd like to move the whole powerset-generating logic [here](https://github.com/elanmart/pytorch/blob/d7103921ca3d3945636e128b2e6884fde21ded98/tools/jit/gen_jit_dispatch.py#L155)?

Reply:
Oh sorry, my bad. I tought I left a different comment here, and didn't bother to read it again 😕 A better solution would be to take a dict that's a mapping from either arg indices or arg names to their new types (and call it e.g. `override_types`). This is similar in implementation, but has much clearer semantics.